engine-tests/src/test/java/org/terasology/metatesting/EntitySystemTest.java

```diff
@@ -0,0 +1,292 @@
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.terasology.engine.context.Context;
+import org.terasology.engine.context.internal.ContextImpl;
+import org.terasology.engine.core.ComponentSystemManager;
+import org.terasology.engine.core.module.ModuleManager;
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.entity.internal.EngineEntityManager;
+import org.terasology.engine.entitySystem.entity.internal.PojoEntityManager;
+import org.terasology.engine.entitySystem.event.AbstractConsumableEvent;
+import org.terasology.engine.entitySystem.event.EventPriority;
+import org.terasology.engine.entitySystem.event.Priority;
+import org.terasology.engine.entitySystem.event.internal.EventSystem;
+import org.terasology.engine.entitySystem.event.internal.EventSystemImpl;
+import org.terasology.engine.entitySystem.systems.BaseComponentSystem;
+import org.terasology.engine.entitySystem.systems.ComponentSystem;
+import org.terasology.engine.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.engine.network.NetworkMode;
+import org.terasology.engine.network.NetworkSystem;
+import org.terasology.engine.recording.RecordAndReplayCurrentStatus;
+import org.terasology.engine.registry.CoreRegistry;
+import org.terasology.engine.registry.In;
+import org.terasology.gestalt.entitysystem.component.Component;
+import org.terasology.gestalt.entitysystem.event.Event;
+import org.terasology.gestalt.entitysystem.event.ReceiveEvent;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class EntitySystemTest {
+
+    private EngineEntityManager entityManager;
+    private EventSystem eventSystem;
+    private Context context;
+
+    @BeforeEach
+    public void setUp() {
+        context = new ContextImpl();
+        CoreRegistry.setContext(context);
+
+        NetworkSystem networkSystem = mock(NetworkSystem.class);
+        when(networkSystem.getMode()).thenReturn(NetworkMode.NONE);
+        context.put(NetworkSystem.class, networkSystem);
+        context.put(RecordAndReplayCurrentStatus.class, new RecordAndReplayCurrentStatus());
+        context.put(ModuleManager.class, mock(ModuleManager.class));
+
+        eventSystem = new EventSystemImpl(networkSystem);
+        context.put(EventSystem.class, eventSystem);
+
+        entityManager = new PojoEntityManager();
+        entityManager.setEventSystem(eventSystem);
+        context.put(EngineEntityManager.class, entityManager);
+        context.put(org.terasology.engine.entitySystem.entity.EntityManager.class, entityManager);
```

Comment:
Can you please clarify why this class reference needs to be fully qualified?

Reply:
It doesn't, lots of places the AI decided to follow this pattern despite me telling it repeatedly not to. Eventually got tired of rejecting those changes individually every time it touched a file, figuring I'll do that in a cleanup-pass and/or try to figure out why it keeps doing it (same with it trying to break lines at just 80 characters wide)